src/pages/Defi/hooks/useFoxyBalances.tsx

```diff
@@ -29,30 +32,35 @@ export function useFoxyBalances() {
       const userAddress = await chainAdapter.getAddress({ wallet, bip44Params })
```

Comment:
danger on the line above

Reply:
Fixed by killing the address-getting from EthereumChainAdapter and removing the default / userAddress logic entirely in profit of making this an exclusively controlled hook